src/build/cargo.rs

```diff
@@ -105,9 +107,15 @@ fn run_cargo(compilation_cx: Arc<Mutex<CompilationContext>>,
 
     let ws = Workspace::new(&manifest_path, &config).expect("could not create cargo workspace");
 
+    // Create and store a build plan for the created Workspace
+    let build_plan = create_plan(&ws).unwrap();
+    compilation_cx_access.build_plan = Some(build_plan);
+    // Release the lock on the compilation_cx
+    mem::drop(compilation_cx_access);
```

Comment:
We definitely could lock twice if that's not a problem, I just tried to avoid that here. Using a scope would need to resurface ws and config, but unsure if the borrows permit that, I'll have to take a look.